### DISCUSSION TITLE
UseFieldArray Form Validation before appending

### DISCUSSION THREAD

**Discussion by embersaw** [react-hook-form/react-hook-form#9615] [Q&A]
![image](https://user-images.githubusercontent.com/63653804/208364305-ad42b8fb-cb5b-406c-b5a9-e780a9b22393.png)

Hello, need help with some validation. As you can see new rows are appended when I click add booking system, i dont want to add new rows unless the previous fields are filled. How to achieve this?

**Comment by bluebill1049** [comment]
I think you will need to check isValid formState, so the append can be prevented before the user proceeds to append the next.

**Reply by embersaw** [reply]
But is there anyway to show the error message?  Also can you please show me a demo in codesandbox?

**Reply by embersaw** [reply]
What I meant was to show the error message only on the dynamic fields not on other fields.

**Reply by bluebill1049** [reply]
couldn't you just check the root of the errors object? 

`{errors.fieldarray && "Please correct error before asset"}`

**Reply by embersaw** [reply]
What I meant was I want to validate only the Dynamic Form Fields in this case name, url, username, password before appending a new booking system and if I click add new booking system before entering something errors should show under those fields only. If I use isValid from formState, it contains the validation result of the entire form and so I cant append a new row unless I enter something in carrier code, carrier name and service field.

**Comment by Moshyfawn** [comment]
Take a look at [the `trigger` method](https://react-hook-form.com/api/useform/trigger)
